packages/x-date-pickers/src/DigitalClock/DigitalClock.tsx

```diff
@@ -348,10 +349,9 @@ export const DigitalClock = React.forwardRef(function DigitalClock<TDate extends
           const formattedValue = utils.format(option, ampm ? 'fullTime12h' : 'fullTime24h');
           const tabIndex =
             focusedOptionIndex === index || (focusedOptionIndex === -1 && index === 0) ? 0 : -1;
-
           return (
             <ClockItem
-              key={formattedValue}
+              key={`${option.valueOf()}-${formattedValue}`}
```

Comment:
Only using `option.valueOf()` fails on `AdapterDayjs` as it handles DST in a strange manner—they don't have duplicate options for the same hour, they format the duplicate offset time differently, but that's why we need to include the `formattedValue` in the key to ensure uniqueness. 🙈 
![Screenshot 2024-10-24 at 12 01 22](https://github.com/user-attachments/assets/a7c8f5ec-7b9e-40ff-a86a-ccee1d969312)

Reply:
:thinking: That is weird indeed

We could also format with the day I guess, but it's probably not worth the computation time that is higher than `.valueOf` +a formatted value that we are also using elsewhere.